### ISSUE TITLE
Exception in Change Feed: org.json.JSONObject$Null cannot be cast to java.util.ArrayList

### ISSUE THREAD

**Issue by jpsimen** [Azure/azure-cosmosdb-spark#256]
I am streaming the change feed from a CosmosDB collection using structured streaming. As soon as a `null` value is added to an array field in the CosmosDB collection, my streaming query crashes with the below stack trace. 

I tried filtering out the null values using various filtering conditions on Spark DataFrames and Datasets, but was not successful. The conversion throwing the exception seems to happen before the filtering. 

Can you please fix that? In the meantime: Is there a workaround?

```
java.lang.ClassCastException: org.json.JSONObject$Null cannot be cast to java.util.ArrayList
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$toSQL$1.apply(CosmosDBRowConverter.scala:113)
	at scala.Option.map(Option.scala:146)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$.toSQL(CosmosDBRowConverter.scala:97)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$1$$anonfun$apply$5.apply(CosmosDBRowConverter.scala:91)
	at scala.Option.map(Option.scala:146)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$1.apply(CosmosDBRowConverter.scala:91)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$1.apply(CosmosDBRowConverter.scala:82)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$.recordAsRow(CosmosDBRowConverter.scala:82)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$asRow$1.apply(CosmosDBRowConverter.scala:68)
	at com.microsoft.azure.cosmosdb.spark.schema.CosmosDBRowConverter$$anonfun$asRow$1.apply(CosmosDBRowConverter.scala:67)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:410)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anonfun$10$$anon$1.hasNext(WholeStageCodegenExec.scala:614)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$$anonfun$run$3.apply(WriteToDataSourceV2.scala:130)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$$anonfun$run$3.apply(WriteToDataSourceV2.scala:129)
	at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1414)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.run(WriteToDataSourceV2.scala:135)
	at org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec$$anonfun$2.apply(WriteToDataSourceV2.scala:79)
	at org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec$$anonfun$2.apply(WriteToDataSourceV2.scala:78)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:345)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Steps for reproducing the bug:

1. Create CosmosDB collection with two documents:
```
  {"id": "1",  "numbers": [1,2,3]}
  {"id": "2",  "numbers": [1,2,3]}
```

2. Start change feed query in Spark

3. In CosmosDD, set one of the arrays to null , for example:
```
  {"id": "1",  "numbers": [1,2,3]}
  {"id": "2",  "numbers": null}
```

**Comment by tknandu** [comment]
Investigating as part of #261 

For structured streaming, a workaround is to set 'InferStreamSchema' to false (documentation [here](https://docs.microsoft.com/en-us/azure/cosmos-db/spark-connector))

**Comment by nomiero** [comment]
This is fixed in 1.3.3 version, can you please upgrade to the latest version and try ?
